### PR TITLE
Use git tag to infer latest version in upgrade test

### LINKED ISSUE #1857: Failure for upgrade test in branch
**Describe the bug**
In #1854 the upgrade test is not using the right latest version function in `test/lib.sh`, because our master does not have the right tag in `git describe`. Once we do this correctly in master from 0.19.0, we should change it back. For now, the upgrade test will always used the latest patch version of 0.18, and CI test will fail if we are committing to older release like 0.17. 

 **Expected behavior**
We should change the function back to use `git describe` to get the target feature branch version.

 **To Reproduce**
Steps to reproduce the behavior.

 **Knative-GCP release version**

 **Additional context**
Add any other context about the problem here such as proposed priority

### PR DESCRIPTION
**Pull request by zhongduo**
Fixes #1857 

## Proposed Changes

-
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: test/lib.sh [hunk 1]
```diff
@@ -63,18 +63,14 @@ function cloud_run_events_setup() {
...
+  local major_minor=$(echo "$semver" | cut -d. -f1-2)
 
-  #local semver=$(git describe --match "v[0-9]*" --abbrev=0)
-  #local major_minor=$(echo "$semver" | cut -d. -f1-2)
-
-  # Get the latest patch release for the major minor
-  #git tag -l "${major_minor}*" | sort -r --version-sort | head -n1
-
-  git tag | sort -r --version-sort | head -n1
+  Get the latest patch release for the major minor
```
Comment (nlopezgi) [inline test/lib.sh:72]: should this be a comment line?

Comment (zhongduo) [reply test/lib.sh]: Good catch! done.

**Pull request merged**